src/store/pageview.ts

```diff
@@ -17,6 +26,29 @@ class ViewStore {
     storage.set(StorageKeys.PAGE_VIEWS, this.pageViews);
   }
 
+  _trim() {
+    this.pageViews.sort((a: PageView, b: PageView): number => b.ts - a.ts);
+    this.pageViews = this.pageViews.filter(
+      (pageView: PageView, i: number) =>
+        pageView.ts > timeStampInSecs() - this.maxAge && i < this.storageSize
+    );
+  }
+
+  setMaxAge(maxAge?: number) {
+    if (!maxAge || maxAge < 0 || maxAge === this.maxAge) return;
+    this.maxAge = maxAge;
+    this._trim();
+    this._save();
+  }
+
+  setStoreSize(storageSize?: number) {
+    if (!storageSize || storageSize < 0 || storageSize === this.storageSize)
+      return;
+    this.storageSize = storageSize;
+    this._trim();
+    this._save();
```

Comment:
As above, I am not sure the API needs two methods, but more importantly this is quite inefficient.

Reply:
Explained above.